modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/core/schema/DataDefinition.es.js

```diff
@@ -81,4 +73,14 @@ export class DataDefinitionSchema extends Schema {
 	get name() {
 		return this[SYMBOL_RAW].name;
 	}
+
+	build() {
+		return {
+			availableLanguageIds: this.availableLanguageIds,
+			contentType: this.contentType,
+			dataDefinitionFields: this.dataDefinitionFields,
+			defaultLanguageId: this.defaultLanguageId,
+			name: this.name,
+		};
+	}
```

Comment:
Why that? this would break with all the lazy implementation here and it also doesn't respect the Schema of the structure.

Reply:
This was used to avoid issues destructing objects: 

```javascript
class DataDefinition {
  get name() {
    return this[SYMBOL_RAW].name;
  }
}

const destructing = {
  ... new DataDefinition(),
}

destructing.name === 'Hello World!'; // false
destructing.name === undefined; // true
``` 

The code is relying a lot on the destructing technique and to take advantage of the lazy load we would have to change a lot more code.

Reply:
Well, I don't see much of an advantage here, because in the end, you do the destructing again in `FieldSetModal`, so we are just doing an unnecessary overhead here, I think the use of the builder is being used in the wrong way here, normally this is used to finalize the action that was done in sequence or in a cascade, where this is not the case here.

For example:
```js
db.table('users').where('name', 'John').first()
```

The `first` here is the builder where the calls to `table` and `where` make up the creation of the query. We don't have that for Schema. I see this in the code:

```js
const {
  availableLanguageIds,
  dataDefinitionFields,
} = dataDefinition.build();

const {dataLayoutPages} = dataLayout.build();

// the same

const {availableLanguageIds, dataDefinitionFields} = dataDefinition;
const {dataLayoutPages} = dataLayout;
```

I can't see a benefit of this here, I understand the problems in relation to destructing but the best for this is to implement a `stringify` feature in Schema where you will only call this when you need to convert the data to `JSON.stringify`.